src/test/scala/is/hail/asm4s/CodeWithRefsSuite.scala

```diff
@@ -0,0 +1,31 @@
+package is.hail.asm4s
+
+import is.hail.check.{Gen, Prop}
+import org.scalatest.testng.TestNGSuite
+import org.testng.annotations.Test
+import is.hail.asm4s.CodeWithRefs._
+import is.hail.asm4s.Code._
+
+object CodeWithRefsSuite {
+  def toISeq(a: Array[Int]): IndexedSeq[Int] = a
+}
+
+class CodeWithRefsSuite extends TestNGSuite {
+
+  @Test def sum() {
+    val m = for {
+      a <- newVar[Array[Int]](Code.newArray[Int](3))
+      _ <- a.update(0, 1)
```

Comment:
We can't use := here?  Oh, there is no ArrayRef.  We should add that.  (Separate PR OK.)